### PR TITLE
refactoring and update schedule list command

### PR DESCRIPTION
**Pull request by Ellrion**

### CODE REVIEW

**Comment by roshangautam** [comment]
Thanks for the pull request but I would love to know the reason behind this refactor.

**Comment by Ellrion** [comment]
Will it be better to split this PR into several different PRs or should I just describe everything that was done in this one?

**Comment by roshangautam** [comment]
No, just want to know why you refactored it. Was there an issue with the code ? or will this refactor increase the performance of the code in some way ?

**Comment by Ellrion** [comment]
1. More clean code
2. Code ready for new PR (fix display command; add sort by next run time etc.)

File: src/Console/Commands/ListSchedule.php [hunk 1]
```diff
@@ -48,41 +49,56 @@ public function __construct(Schedule $schedule)
...
+    protected function getCommandNextRunTime(Event $event)
     {
         $date = Carbon::now();
 
         if ($event->timezone) {
             $date->setTimezone($event->timezone);
         }
 
-        return (CronExpression::factory($event->expression)->getNextRunDate($date->toDateTimeString()))->format('Y-m-d H:i:s');
+        return CronExpression::factory($event->getExpression())->getNextRunDate($date->toDateTimeString());
```
Comment (roshangautam) [inline src/Console/Commands/ListSchedule.php:102]: I dont think you need to make a call to getExpression here. `$event` object automatically handles that.

**Comment by roshangautam** [comment]
Are you trying to say you're gonna send another pull request which will sort scheduled commands by upcoming schedule datetime ? because I dont see anything like that in this pull request.

File: src/Console/Commands/ListSchedule.php [hunk 2]
```diff
@@ -48,41 +49,56 @@ public function __construct(Schedule $schedule)
...
+        });
+
+        $this->table(
+            ['Description', 'Command', 'Schedule', 'Upcoming', 'Timezone', 'Overlaps?', 'In Maintenance?'],
+            $events
+        );
+    }
+
+    /**
+     * Possible change command text for simple view.
```
Comment (roshangautam) [inline src/Console/Commands/ListSchedule.php:78]: Change this to "Remove parameters, if any, from command string"

File: src/Console/Commands/ListSchedule.php [hunk 3]
```diff
@@ -48,41 +49,56 @@ public function __construct(Schedule $schedule)
...
     /**
-     * Get Upcoming schedule.
+     * Get next time command run datetime.
      *
-     * @return bool
+     * @param Event $event
+     * @return \DateTime
      */
-    protected function upcoming($event)
+    protected function getCommandNextRunTime(Event $event)
```
Comment (roshangautam) [inline src/Console/Commands/ListSchedule.php:94]: Please change the function name to `getUpcomingScheduledDateTime` and change phpdoc description to `Get Upcoming scheduled date time for the given event`

**Review by roshangautam — COMMENTED** [review]
Please view comments

File: src/Console/Commands/ListSchedule.php (hunk 2 above)
Comment (Ellrion) [reply src/Console/Commands/ListSchedule.php:78]: In this method we can add different logic for schedule command text prepare. Not only "Remove parameters, if any, from command string"

File: src/Console/Commands/ListSchedule.php (hunk 3 above)
Comment (Ellrion) [reply src/Console/Commands/ListSchedule.php:94]: In method we call `getNextRunDate` and we have method `prepareCommandToDisplay`(prepare**Command**ToDisplay). The name `getCommandNextRunDateTime` more consistently.

File: src/Console/Commands/ListSchedule.php (hunk 1 above)
Comment (Ellrion) [reply src/Console/Commands/ListSchedule.php:102]: methods a better  than properties becouse property can become protection, becouse exists public getter

**Comment by Ellrion** [comment]
> you're gonna send another pull request which will sort scheduled commands by upcoming schedule datetime 

see https://gist.github.com/Ellrion/1a3ef7bc2e988e3fb80782e66d363fb7#file-schedulelistcommand-php
and screenshots in comment

**Comment by roshangautam** [comment]
Can you please send another pull request with the modifications shown in your gist ? I am gonna close this for now. Also please provide a description while sending a pull request.

**Pull request closed without merging**